docs/general/attributes.md

```diff
@@ -209,6 +142,46 @@ if they do not cause breaking changes to HTTP semantic conventions.
 | `ipv6` | IPv6 |
 <!-- endsemconv -->
 
+### `network.peer.*` and `network.local.*` attributes
+
+These attributes identify network peers that are directly connected to each other.
+
+For IP-based communication, `network.peer.address` and `network.local.address` should be IP addresses, Unix domain socket names, or other addresses specific to network type.
+
+Semantic conventions and instrumentations that populate `address`/`port` pairs
+on `server.*`, `client.*`, `destination.*` or `source.*`, SHOULD only populate the `address`/`port` pair on
+`network.peer.*` and `network.local.*` when they are not the same.
+
+_Note: Specific structures and methods to obtain socket-level attributes are mentioned here only as examples. Instrumentations would usually use Socket API provided by their environment or sockets implementations._
+
+When connecting using `connect(2)`
+on [Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/connect.2.html) or [Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-connect)
+with `AF_INET` address family, `network.peer.address` and `network.peer.port` represent `sin_addr` and `sin_port` fields of [`sockaddr_in`](https://man7.org/linux/man-pages/man7/ip.7.html) structure.
+
+When using `bind(2)`
+on [Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/bind.2.html) or [Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-bind)
+with `AF_INET` address family, `network.peer.address` and `network.peer.port` represent `sin_addr` and `sin_port` fields of `sockaddr_in` structure.
```

Comment:
nit:
This sentences are very similar, maybe it's possible to merge them? E.g. 

```suggestion
`When connecting using `connect(2)` ([Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/connect.2.html) or [Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-connect)) or `bind(2)`([Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/bind.2.html) or [Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-bind)) with `AF_INET` address family, `network.peer.address` and `network.peer.port` represent `sin_addr` and `sin_port` fields of `sockaddr_in` structure.
```

?